Sources/SwiftDocC/Model/Identifier.swift

```diff
@@ -141,10 +141,14 @@ extension TopicReferenceResolutionErrorInfo {
 /// > its data.
 public struct ResolvedTopicReference: Hashable, Codable, Equatable, CustomStringConvertible {
     typealias ReferenceBundleIdentifier = String
-    typealias ReferenceKey = String
+    private struct ReferenceKey: Hashable {
+        var path: String
+        var fragment: String?
+        var sourceLanguages: Set<SourceLanguage>
```

Comment:
Should we write an initializer and validate this is not an empty set? Or that the path is not empty or invalid in some other way?

Reply:
Would it make sense to conform to CustomDebugStringConvertible and return the string we used previously below in `cacheKey`? Seems like it would be a readable format for debugging purposes.

Reply:
As a key it doesn't matter if the path or the set is empty. The main `ResolvedTopicReference` initializer will perform any validation that's relevant to the reference but the key only needs to be a hash-able container for the original values.